File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // Use a hash map 
        unordered_map<int, int> targetMap; 

        for(int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i]; 
            if(targetMap.find(diff) != targetMap.end()) {
                return {targetMap[diff], i};
            }
            targetMap.insert({nums[i], i});
        } 

        return {};
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.cs
public class Solution {

    public int BinarySearch(int l, int r, int[] nums, int target) {
        if(l > r) return -1; 

        int m = l + (r - l) / 2;

        if(nums[m] > target) {
            return BinarySearch(l, m-1, nums, target);
        }
        else if(nums[m] < target) {
            return BinarySearch(m+1, r, nums, target);
        }
        else {
            return m;
        }
    }

    public int Search(int[] nums, int target) {
        return BinarySearch(0, nums.Length-1, nums, target);
    }
}
